modules/mono/editor/bindings_generator.cpp

```diff
@@ -730,7 +730,8 @@ Error BindingsGenerator::_generate_cs_type(const TypeInterface &itype, const Str
 	}
 
 	output.push_back(INDENT1 "public ");
-	output.push_back(itype.is_singleton ? "static class " : "class ");
+	bool is_abstract = !ClassDB::can_instance(itype.name) && ClassDB::is_class_enabled(itype.name); // can_instance returns true if there's a constructor and the class is not 'disabled'
+	output.push_back(itype.is_singleton ? "static class " : (is_abstract ? "abstract class " : "class "));
```

Comment:
Hmm, I have the feeling I had a good reason not to make these abstract, but I don't remember... 🤷‍♂️ 
BTW, this should use `itype.is_instantiable` instead.

Reply:
Hi, the reason I didn't use `itype.is_instantiable` is described in the comment.  
It would make the `RID` and `NodePath` classes abstract because they are set as `disabled` in `ClassDB`. Here, my condition only check for the absence of constructor. It is essentially `!itype.is_instantiable && ClassDB::is_class_enabled(itype.name)`.  
I first thought of using directly `itype.is_instantiable` and setting `RID` and `NodePath` as instantiable but think it'll have a greater chance of have side effects.

Reply:
I see. I plan to move NodePath and RID to cs_files since we already maintain the internal calls manually. I will change this when that happens.

Reply:
>Hmm, I have the feeling I had a good reason not to make these abstract, but I don't remember...

2 and a half years later, I now remember. Some Godot classes are registered as virtual/non-instantiable, but they're still instantiable by the engine and instances of them are returned to the scripting API. An example of this is `SceneTreeTimer`, returned from `SceneTree.create_timer`.

I think the reason this wasn't resulting in any errors is because the Mono embedding API seems to allow us to allocate abstract classes. Soon we will change the way we do this though, so it will result in either compiler error, type initialization exception or invalid cast exception.

I think I'll have to change it back, while keeping the constructor internal. What was the motivation for this change? Do you think removing `abstract` could cause problems?

Reply:
This was only motivated by user experience and usability issues. Basically: intellisense would tell you you could instance a class, when in reality it was abstract and you couldn't.  
I don't see how removing it would cause any problem, no.

Reply:
Thanks!